### ISSUE TITLE
Garden server fails silently on port conflicts

### ISSUE THREAD

**Issue by eysi09** [garden-io/garden#5158]
### Current Behavior

If I run `garden dev` in a given terminal window it starts a server on the default port, `9777`. 

Now, I I run another `garden dev` command in a different window it also the picks the same port. It attempts to start the server at that port which fails silently. 

The error is never logged and Garden continues running, so to the user, everything looks okay. However, in this state, the web dashboard won't work.

### Expected behavior

Firstly, that Garden picks the next available port if the current one is in use. The library we use for picking ports doesn't appear to do that and we may need to implement this on our end.

Second, that Garden doesn't just swallow these errors. Here's the offending code: https://github.com/garden-io/garden/blob/main/core/src/server/server.ts#L184

If we were to log the error there, we'd see something like:

```console
Error Error: listen EADDRINUSE: address already in use ::1:9777
    at Server.setupListenHandle [as _listen2] (node:net:1740:16)
    at listenInCluster (node:net:1788:12)
    at GetAddrInfoReqWrap.doListen [as callback] (node:net:1937:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:8)
    at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'EADDRINUSE',
  errno: -48,
  syscall: 'listen',
  address: '::1',
  port: 9777
}
``` 

Instead this information is just swallowed and the user thinks that Garden is running as expected. 

### Reproducible example

- Add a console.log to [this callback](https://github.com/garden-io/garden/blob/main/core/src/server/server.ts#L184) that logs the error
- Run `garden dev` in one terminal window and observe how it starts up without isses
- Run `garden dev` in a second terminal window and observe how the error is logged but that Garden otherwise continues running and that you wouldn't know the server had issues without that console.log.

### Workaround

Manually set the port when multiple Garden dev commands. You can e.g. run the first one with `garden dev`, second one with `garden dev --port 9778`, etc. 

### Suggested solution(s)

See "Expected behaviour" above.

### Your environment

- macOS M1, Monterey
- Latest main

**Comment by vvagaytsev** [comment]
@eysi09 should we revert the changes from #4984 and reopen #4969?
An ugly error is better than a silent failure. That can be a quick fix. Later we can implement a proper solution to resolve both issues.

**Comment by eysi09** [comment]
Ideally we just fix the underlying issue. It shouldn't be that much work. 

Worst case we can catch the `EADDRINUSE` error and try with a new port, if the `get-port-please` library doesn't work.

**Comment by vvagaytsev** [comment]
Actually, `get-port-please` work fine, it was just misused in #4984. I fixed that in #5163.
